plugins/optimization-detective/detection.php

```diff
@@ -35,8 +35,24 @@ function od_get_detection_script( string $slug, OD_URL_Metrics_Group_Collection
 	 */
 	$detection_time_window = apply_filters( 'od_detection_time_window', 5000 );
 
-	$web_vitals_lib_data = require __DIR__ . '/detection/web-vitals.asset.php';
-	$web_vitals_lib_src  = add_query_arg( 'ver', $web_vitals_lib_data['version'], plugin_dir_url( __FILE__ ) . '/detection/web-vitals.js' );
+	try {
+		$web_vitals_lib_data = require __DIR__ . '/detection/web-vitals.asset.php';
+	} catch ( Error $error ) {
+		$error_message  = '[Optimization Detective] ';
+		$error_message .= sprintf(
+			/* translators: 1: File path. 2: CLI command. */
+			__( 'Unable to load %1$s. Please make sure you have run %2$s.', 'optimization-detective' ),
+			'detection/web-vitals.asset.php',
+			'`npm install && npm run build:optimization-detective`'
+		);
+		if ( WP_DEBUG && WP_DEBUG_DISPLAY ) {
+			$error_message .= ' ' . __( 'PHP Error:', 'optimization-detective' ) . ' ' . $error->getMessage();
+		}
+		return wp_get_inline_script_tag(
+			sprintf( 'console.error( %s );', wp_json_encode( $error_message ) )
+		);
+	}
+	$web_vitals_lib_src = add_query_arg( 'ver', $web_vitals_lib_data['version'], plugin_dir_url( __FILE__ ) . '/detection/web-vitals.js' );
```

Comment:
Shouldn't this rather be a check in the plugin main file? If that file is missing, the plugin needs to be built, just how e.g. other plugins commonly check for existence of `composer.json` or `vendor` dir. The lack of this should be obvious to the end user, IMO better in an admin notice that shows immediately rather than a hard-to-detect PHP error that is only shown in JavaScript and only if the detection script is attempted to be loaded.

Reply:
This is the only place where the file is actually accessed. Also, the check is only relevant during development. It seems wasteful to do a `file_exists()` check with every request in production when in a production build the files are always present.

Reply:
> Also, the check is only relevant during development.

So is checking for things like `composer.json` or `vendor`, but it's commonly included in many plugins' main file. My main feedback here is that I don't see why this file missing should not be treated in the same way as such cases. It's needed for the plugin to function correctly, as far as I understand.

Reply:
OK. Reverted in 6ae5fb01cca13b0f1a2f1d3de1859debf65f3ccd

Reply:
Moved to main file in 3aae425a62410655d585f0a1d629719a7d4d39e2